docs/grammar.txt

```diff
@@ -32,9 +32,14 @@ EventDefinition = 'event' Identifier IndexedParameterList 'anonymous'? ';'
 EnumValue = Identifier
 EnumDefinition = 'enum' Identifier '{' EnumValue? (',' EnumValue)* '}'
 
-IndexedParameterList = '(' ( TypeName 'indexed'? Identifier? (',' TypeName 'indexed'? Identifier?)* )? ')'
-ParameterList =        '(' ( TypeName            Identifier? (',' TypeName            Identifier?)* )? ')'
-TypeNameList =         '(' ( TypeName (',' TypeName )* )? ')'
+IndexedParameterList = '(' ( IndexedParameter (',' IndexedParameter )* )? ')'
+IndexedParameter = TypeName 'indexed'? Identifier?
```

Comment:
Sorry, I think as a separate change it would make more sense renaming `IndexedParameter*` to `EventParameter*` because it reflects more what it is. Indexed would suggest the parameters are indexed themselves (as opposed to a modifier).

Reply:
Good. Should I rename TypeNameList and UnnamedParamete to FunctionTypeParameterList and FunctionTypeParameter?

Reply:
If it is only used in function types then it would make sense.